src/java.base/share/classes/java/nio/Direct-X-Buffer.java.template

```diff
@@ -528,6 +530,21 @@ class Direct$Type$Buffer$RW$$BO$
     }
 #end[char]
 
+#if[byte]
+    @Override
+    int scaleFactor() {
+        return 0;
+    }
+
+    @Override
+    AbstractMemorySegmentImpl arrayBackedSegment(Object base,
+                                                 long offset,
+                                                 long length,
+                                                 boolean readOnly,
+                                                 MemorySessionImpl bufferScope) {
+        return SegmentFactories.arrayOf$Type$Segment(base, offset, length, readOnly, bufferScope);
```

Comment:
For direct buffers I don't think we can get here since the base should be `null`? I think this should be changed to throw UOE (if we ever do end up here, we don't want to return a heap segment with a `null` base).